wasm/src/main/scala/ir2wasm/WasmExpressionBuilder.scala

```diff
@@ -305,11 +306,111 @@ private class WasmExpressionBuilder private (
           IRTrees.ApplyStatically(t.flags, t.receiver, className, t.method, t.args)(t.tpe)(t.pos)
         )
 
+      case IRTypes.AnyType if t.method.name.isReflectiveProxy =>
+        genReflectiveCall(t)
+
       case _ =>
         genApplyNonPrim(t)
     }
   }
 
+  private def genReflectiveCall(t: IRTrees.Apply): IRTypes.Type = {
+    // A local for a copy of the receiver that we will use to resolve dispatch
+    val receiverLocalForDispatch =
+      fctx.addSyntheticLocal(Types.WasmRefType.any)
+    val reflectiveProxies =
+      fctx.addSyntheticLocal(Types.WasmRefType(WasmArrayTypeName.reflectiveProxies))
+    val size = fctx.addSyntheticLocal(Types.WasmInt32)
+    val i = fctx.addSyntheticLocal(Types.WasmInt32)
+
+    // taking a hashCode for ease, but it might be better to compare in string
+    // to completely get rid of the risk of hash collision(?)
+    val methodName = t.method.name.nameString.hashCode()
```

Comment:
That's definitely not going to be safe. Collisions on `String.hashCode` are notoriously not very hard to produce.

But it's OK; we have the whole program. We can give a unique integer to every `MethodName.isReflectiveProxy` that we encounter in the program. ;)